kong/clustering/version_negotiation/services_known.lua

```diff
@@ -0,0 +1,6 @@
+return {
+  cluster_protocol = {
```

Comment:
what does the `services_known` mean here? My understanding is that this file really just defines the different endpoints along with the different protocols they support. `cluster` is not really a `service` but an endpoint. In the `wrpc` mode, it can actually be used for the communications of multiple `services` defined in https://github.com/Kong/kong/pull/8357/files#diff-82fded69f85ff8ba289c5a4e082e55791e36b462a43e2465f50848513a3187ee

Reply:
I guess in the existing setup, `endpoint` and `service` has one to one mapping, so it is fined to use them interchangeably. However, it won't be the case anymore once we start using wrpc as the underlying protocol. Is it correct?

Reply:
yes, I understand how wrpc is intended to support several sets of methods, calling each one a "service".  the main use for this version negotiation will be to determine which of these are supported on both sides; but could be anything that we make optional or multi-versioned.

this `services_known.lua` file is the list of services the CP supports, and the order of versions on each one determines the priority (if more than one version is supported on both sides, the first one in this file "wins").

The content here is just the minimal example to make the system work, so I decided to use the most basic part of DP/CP communication: the protocol itself, which we will have to support at least two versions soon (the current json-based and the wrpc in #8357)

Reply:
Cool. I was just trying to understand what additional changes I would need to make to support the `analytics` service we were going to introduce in kong 3.0. https://github.com/Kong/kong-dp-spec/blob/c83273a7eb10c8960c95f415c5882e62748e112c/spec/proto/kong/services/analytics/v1/analytics.proto

Reply:
I wasn't sure whether I needed to update this file or not.

Reply:
yes, just add a new service like
```lua
...
  analytics = {
    { version = "v1", message = "first!" },
  },
...
```

also, since protobuf makes it simple to add fields, and everything is optional, in many cases it "should" be possible to extend services without having to declare new versions.  my guess is that it would be necessary only if we decide to change an expected sequence of calls, or make a method not only obsolete but inconvenient.  i hope in most cases the trick would be to carefully choose the default case semantics.